pkg/controllers/controllers.go

```diff
@@ -188,7 +188,8 @@ func Register(ctx context.Context, systemNamespace string, cfg clientcmd.ClientC
 		appCtx.BundleDeployment(),
 		appCtx.GitRepoRestriction().Cache(),
 		appCtx.Bundle(),
-		appCtx.GitRepo())
+		appCtx.GitRepo(),
+		appCtx.Core.Secret().Cache())
```

Comment:
Why cache the secret?

Reply:
This avoids api call which cause latency